wallet_test.go

```diff
@@ -0,0 +1,393 @@
+			"method": "fixed",
+			"started_at": null,
+			"expiration_at": "2026-12-31T23:59:59Z",
+			"status": "active",
+			"target_ongoing_balance": "0.00",
+			"threshold_credits": "0.00",
+			"trigger": "interval",
+			"paid_credits": "105.00",
+			"granted_credits": "105.00",
+			"created_at": "2022-04-29T08:59:51Z",
+			"invoice_requires_successful_payment": false,
+			"transaction_metadata": [],
+			"transaction_name": "Recurring Transaction Rule"
+		}],
+		"applies_to": {
+			"fee_types": ["charge"],
+			"billable_metric_codes": ["bm1"]
+		}
+	}
+}`
+
+var mockWalletListResponse = `{
+	"wallets": [{
+		"lago_id": "b1b2c3d4-e5f6-7890-1234-56789abcdef0",
+		"lago_customer_id": "a1a2b3b4-c5d6-7890-1234-56789abcdef0",
+		"external_customer_id": "12345",
+		"status": "active",
+		"currency": "USD",
+		"name": "wallet name",
+		"rate_amount": "1.00",
+		"credits_balance": "100.00",
+		"balance_cents": 10000,
+		"expiration_at": "2022-07-07T23:59:59Z",
+		"created_at": "2022-04-29T08:59:51Z",
+		"recurring_transaction_rules": [{
+			"lago_id": "c1c2d3d4-e5f6-7890-1234-56789abcdef0",
+			"interval": "monthly",
+			"method": "fixed",
+			"started_at": null,
+			"expiration_at": "2026-12-31T23:59:59Z",
+			"status": "active",
+			"target_ongoing_balance": "0.00",
+			"threshold_credits": "0.00",
+			"trigger": "interval",
+			"paid_credits": "105.00",
+			"granted_credits": "105.00",
+			"created_at": "2022-04-29T08:59:51Z",
+			"invoice_requires_successful_payment": false,
+			"transaction_metadata": [],
+			"transaction_name": "Recurring Transaction Rule"
+		}],
+		"applies_to": {
+			"fee_types": ["charge"],
+			"billable_metric_codes": ["bm1"]
+		}
+	}],
+	"meta": {
+		"current_page": 1,
+		"next_page": 2,
+		"prev_page": null,
+		"total_pages": 7,
+		"total_count": 63
+	}
+}`
+
+func TestWallet_Create(t *testing.T) {
+	t.Run("When the server is not reachable", func(t *testing.T) {
+		c := qt.New(t)
+
+		client := New().SetBaseURL("http://localhost:88888").SetApiKey("test_api_key")
+		result, err := client.Wallet().Create(context.Background(), &WalletInput{
+			ExternalCustomerID: "12345",
+			RateAmount:         "1.00",
+			Name:               "wallet name",
+			PaidCredits:        "100.00",
+			GrantedCredits:     "100.00",
+			ExpirationAt:       Ptr(time.Date(2022, 7, 7, 23, 59, 59, 0, time.UTC)),
+			RecurringTransactionRules: []RecurringTransactionRuleInput{
+				{
+					PaidCredits:      "105.00",
+					GrantedCredits:   "105.00",
+					ThresholdCredits: "0.00",
+					Trigger:          "interval",
+					Interval:         "monthly",
+					Method:           "fixed",
+					StartedAt:        nil,
+					ExpirationAt:     Ptr(time.Date(2026, 12, 31, 23, 59, 59, 0, time.UTC)),
+					TransactionName:  "Recurring Transaction Rule",
+				},
+			},
+			AppliesTo: AppliesTo{
+				FeeTypes:            []string{"charge"},
+				BillableMetricCodes: []string{"bm1"},
+			},
+		})
+
+		c.Assert(err, qt.IsNotNil)
+		c.Assert(result, qt.IsNil)
```

Comment:
does it works because it expects `localhost:8888` does not have a server listening and will return a not reachable server?